# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Stencil.h
# Commit: 8af30c6d9bb9
# Full Hash: 8af30c6d9bb91a1ed9a5bee712252bbe1bd20d85
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 11: Use TaggedParserAtomIndex in ScriptThingVariant. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96918
# ==============================================================================

diff -r cb0e58368cd4 -r 8af30c6d9bb9 js/src/frontend/Stencil.h
--- a/js/src/frontend/Stencil.h	Sat Nov 14 00:03:28 2020 +0000
+++ b/js/src/frontend/Stencil.h	Sat Nov 14 00:03:36 2020 +0000
@@ -424,13 +424,11 @@
 // convert to a nullptr.
 class NullScriptThing {};
 
-using ScriptAtom = const ParserAtom*;
-
 // These types all end up being baked into GC things as part of stencil
 // instantiation.
 using ScriptThingVariant =
-    mozilla::Variant<ScriptAtom, NullScriptThing, BigIntIndex, ObjLiteralIndex,
-                     RegExpIndex, ScopeIndex, FunctionIndex,
+    mozilla::Variant<TaggedParserAtomIndex, NullScriptThing, BigIntIndex,
+                     ObjLiteralIndex, RegExpIndex, ScopeIndex, FunctionIndex,
                      EmptyGlobalScopeType>;
 
 // Data generated by frontend that will be used to create a js::BaseScript.